internal/stanza/converter_test.go

```diff
@@ -69,8 +113,94 @@ func complexEntry() *entry.Entry {
 	return e
 }
 
-func TestConvertMetadata(t *testing.T) {
+func TestAllConvertedEntriesAreSentAndReceived(t *testing.T) {
+	testcases := []struct {
+		entries            int
+		flushTriggerCount  uint
+		flushInterval      time.Duration
+		expectedNumFlushes int
+	}{
+		{
+			entries:            10,
+			flushTriggerCount:  10,
+			flushInterval:      5 * time.Millisecond,
+			expectedNumFlushes: 1,
+		},
+		{
+			entries:            10,
+			flushTriggerCount:  3,
+			flushInterval:      5 * time.Millisecond,
+			expectedNumFlushes: 4,
+		},
+		{
+			entries:            100,
+			flushTriggerCount:  20,
+			flushInterval:      5 * time.Millisecond,
+			expectedNumFlushes: 4,
+		},
+	}
+
+	for i, tc := range testcases {
+		tc := tc
+
+		t.Run(strconv.Itoa(i), func(t *testing.T) {
+			converter := NewConverter(
+				WithFlushTriggerCount(tc.flushTriggerCount),
+				WithFlushInterval(tc.flushInterval),
+			)
+			converter.Start()
+			defer converter.Stop()
+
+			go func() {
+				for _, ent := range complexEntries(tc.entries) {
+					assert.NoError(t, converter.Batch(ent))
```

Comment:
I don't think this is testing the `flushInterval` in any cases. Perhaps we need to add a sleep in this loop, and ensure that in some test cases, the number of entries * sleep duration exceeds the flush interval.

Reply:
I've added a couple of tests for the flushInterval in `TestAllConvertedEntriesAreSentAndReceivedWithinAnExpectedTimeDuration` but my concern is that those are time based tests which might end up being flaky (depending on the load on the machine that the test is running on).

Feel free to comment the approach.